config/src/tests/failover/failover.cpp

```diff
@@ -106,22 +108,21 @@ struct ServerFixture {
 
     void start()
     {
-        supervisor = new FRT_Supervisor();
+        frtServer = std::make_unique<fnet::frt::StandaloneFRT>();
+        supervisor = & frtServer->supervisor();
         server.init(supervisor);
         supervisor->Listen(get_port(listenSpec));
         wait(); // Wait until test runner signals we can start
-        supervisor->Main();
         wait(); // Signalling that we have shut down
         wait(); // Wait for signal saying that supervisor is deleted
     }
 
     void stop()
     {
-        if (supervisor != NULL) {
-            supervisor->ShutDown(true);
+        if (frtServer) {
             wait(); // Wait for supervisor to shut down
-            delete supervisor;
-            supervisor = NULL;
+            frtServer.reset();
```

Comment:
wait is now called before shutting down instead of after

Reply:
Clean up and fixed order.